apps/scratchpad/src/createAccount.ts

```diff
@@ -64,24 +69,52 @@ export async function createAccount() {
     return sigHex;
   };
   const dryRun = false;
+
+  const pubKey = Buffer.from(pubKeyHex.substring(56), "hex");
+  if (pubKey.length !== 64) {
+    throw new Error("Invalid public key, wrong length");
+  }
+
+  const key1 = `0x${pubKey.subarray(0, 32).toString("hex")}` as Hex;
+  const key2 = `0x${pubKey.subarray(32).toString("hex")}` as Hex;
+  const salt = 0n;
+  const args = [[key1, key2], salt] as const;
+
+  const address = await publicClient.readContract({
+    ...accountFactoryConfig,
+    functionName: "getAddress",
+    args,
+  });
+
+  const funderPrivKey = `0x${process.env.DAIMO_API_PRIVATE_KEY}` as const;
+  const funderAccount = privateKeyToAccount(funderPrivKey);
+  console.log(`Faucet account: ${funderAccount.address}`);
+  const walletClient = createWalletClient({
+    account: funderAccount,
+    chain,
+    transport: http(),
+  });
+
+  // Deploy account
+  const hash = await walletClient.writeContract({
+    ...accountFactoryConfig,
+    functionName: "createAccount",
+    args,
+  });
+  console.log(`[API] deploy transaction ${hash}`);
+  const tx = await publicClient.waitForTransactionReceipt({ hash });
+  console.log(`[API] deploy transaction ${tx.status}`);
+
   const account = await DaimoAccount.init(
     publicClient,
-    pubKeyHex,
+    address,
     signer,
     dryRun
   );
   const addr = account.getAddress();
   console.log(`Burner Daimo account: ${addr}`);
 
   // Fund it from the faucet
-  const faucetPrivKey = `0x${process.env.DAIMO_API_PRIVATE_KEY}` as const;
-  const faucetAccount = privateKeyToAccount(faucetPrivKey);
-  console.log(`Faucet account: ${faucetAccount.address}`);
-  const walletClient = createWalletClient({
-    account: faucetAccount,
-    chain,
-    transport: http(),
-  });
   const usdcTxHash = await walletClient.writeContract({
```

Comment:
step 3 and 4: send USDC to account, prefund EntryPoint, each with a sequential delay of a few seconds for inclusion

fine for the scratchpad. i agree re: doing it in a single transaction in-app